app/src/main/java/com/babylon/wallet/android/presentation/onboarding/restore/mnemonics/RestoreMnemonicsViewModel.kt

```diff
@@ -57,39 +58,13 @@ class RestoreMnemonicsViewModel @Inject constructor(
             val factorSources = when (args) {
                 is RestoreMnemonicsArgs.RestoreProfile -> {
                     val profile = getTemporaryRestoringProfileForBackupUseCase(args.backupType)?.changeGateway(Radix.Gateway.mainnet)
-                    val allAccounts = profile?.currentNetwork?.accounts.orEmpty()
-
-                    profile?.factorSources
-                        ?.filterIsInstance<DeviceFactorSource>()
-                        ?.filter { !mnemonicRepository.mnemonicExist(it.id) }
-                        ?.mapNotNull { factorSource ->
-                            val associatedAccounts = allAccounts.filter { it.factorSourceId() == factorSource.id }
-
-                            if (associatedAccounts.isEmpty() && !factorSource.isBabylon) return@mapNotNull null
-
-                            RecoverableFactorSource(
-                                associatedAccounts = associatedAccounts,
-                                factorSource = factorSource
-                            )
-                        }
-                        .orEmpty()
+                    profile.recoverableFactorSources()
                 }
 
+                // TODO refactor later to remove that specific case at all and use RestoreProfile? always
```

Comment:
I prefer not to refactor this now, and leave it to tackle with next release and Im happy to continue with that to use only one arg type with navType and isMandatory instead of RestoreSpecificMnemonic and RestoreProfile